wallet/wallet-cli-lib/src/commands/mod.rs

```diff
@@ -658,14 +658,15 @@ impl CommandHandler {
             }
 
             WalletCommand::GenerateBlock { transactions } => {
-                let transactions_opt =
-                    transactions.map(|txs| txs.into_iter().map(HexEncoded::take).collect());
+                let transactions = transactions.into_iter().map(HexEncoded::take).collect();
                 let block = controller_opt
                     .as_mut()
                     .ok_or(WalletCliError::NoWallet)?
                     .generate_block(
                         selected_account.ok_or(WalletCliError::NoSelectedAccount)?,
-                        transactions_opt,
+                        transactions,
+                        vec![],
+                        false,
```

Comment:
Unless I'm misunderstanding something, this seems to be a big, big mistake... we're basically blocking the main staking mechanism from using the mempool.

Reply:
Ah, good pick up. Thanks you're right.